src/background/injector/content-script-injector.ts

```diff
@@ -10,31 +9,26 @@ export class ContentScriptInjector {
     public static readonly cssFiles: string[] = ['injected/styles/default/injected.css', 'bundle/injected.css'];
 
     public static timeoutInMilliSec = 5e4;
-    private readonly _chromeAdapter: BrowserAdapter;
-    private readonly _q: typeof Q;
-
-    constructor(chromeAdapter: BrowserAdapter, q: typeof Q) {
-        this._chromeAdapter = chromeAdapter;
-        this._q = q;
-    }
 
-    public injectScripts(tabId: number): Q.IPromise<null> {
-        const deferred = this._q.defer<null>();
+    constructor(private readonly chromeAdapter: BrowserAdapter, private readonly promiseFactory: PromiseFactory) {}
 
-        ContentScriptInjector.cssFiles.forEach(file => {
-            this._chromeAdapter.injectCss(tabId, file, null);
-        });
+    public injectScripts(tabId: number): Promise<null> {
+        const inject = new Promise<null>((resolve, reject) => {
+            ContentScriptInjector.cssFiles.forEach(file => {
+                this.chromeAdapter.injectCss(tabId, file, null);
```

Comment:
It's a good question! There was a reason when we implemented this but I don't recall. Maybe @waabid remember?

Reply:
I guess the reason here is that, we need js to be injected in all frames before we start sending messages. but, we don't have to wait for css, since css loading doesn't affect our messaging.